lib/gosu/compat.rb

```diff
@@ -103,24 +103,26 @@ def self.from_text(*args)
   end
 end
 
-# No need to pass a Window to Sample.
-class Gosu::Sample
-  alias initialize_without_window initialize
-
+# No need to pass a Window to Sample and even better don't use Sample at all.
+class Gosu::Sample < Gosu::Sound
   def initialize(*args)
+    Gosu.deprecation_message("Gosu::Sample has been deprecated in Gosu 0.15; use Sound with streaming = false instead, see https://www.libgosu.org/rdoc/Gosu/Sound.html.")
+
     if args.first.is_a? Gosu::Window
       args.shift
       Gosu.deprecation_message("Passing a Window to Sample#initialize has been deprecated in Gosu 0.7.17.")
     end
-    initialize_without_window(*args)
+    super(*args, strea)
   end
 end
 
-# No need to pass a Window to Song.
+# No need to pass a Window to Song and even better don't use Song at all.
 class Gosu::Song
```

Comment:
Depending on whether or not the c++ counterpart of Song should still exists this might need inherit from Gosu::Sound too.